src/raftstore/store/peer.rs

```diff
@@ -453,21 +507,32 @@ impl Peer {
             }
         }
 
-        if need_repropose {
-            try!(self.repropose_pending_cmds());
-        }
-
         Ok(results)
     }
 
-    fn handle_raft_entry_normal(&mut self,
-                                entry: &raftpb::Entry,
-                                repropose: &mut bool)
-                                -> Result<Option<ExecResult>> {
+    fn repropose_conf_change(&mut self) {
+        while let Some(mut cc) = self.pending_cmds.pop_conf_change() {
+            let res = self.propose_conf_change(cc.cmd.take().unwrap());
```

Comment:
Which `take` do you mean? `repropose_conf_change` is called after last conf change is processed, so the head of the queue should have cmd.